src/application/estimator/model/model_dynamics.c

```diff
@@ -0,0 +1,168 @@
+#include "application/estimator/model/model_dynamics.h"
+#include "application/estimator/estimator_types.h"
+#include "application/estimator/model/model_airdata.h"
+#include "application/estimator/model/quaternion.h"
+#include "application/logger/log.h"
+#include "common/math/math-algebra3d.h"
+#include "common/math/math.h"
+#include "drivers/timer/timer.h"
+#include <math.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+/*
+ * Parameters
+ * go here as static const, or as define
+ */
+// aerodynamic parameters
+static const float c_canard =
+    (2 * (4 * 0.0254) * (2.5 * 0.0254)) *
+    (0.203 / 2 + 0.0254); // moment arm * area of canard 
+static const float cn_alpha = 5.0f; // pitch forcing coeff
+static const float cn_omega = 0.0f; // roll damping coeff
+static const float area_reference = M_PI * powf((0.203 / 2), 2); // cross section of body tube
+// c_aero = area_reference * (length_cp-length_cg), center of pressure(cp): -0.5, center of
+// gravity(cg): 0
+static const float c_aero = area_reference * (-0.5f);
+
+// mass and inertia
+static const matrix3d_t inertia_matrix = {
+    .array = {{0.17, 0, 0}, {0, 11.0, 0}, {0, 0, 11.0}}
+}; // inertia matrix of the rocket (J in matlab)
+static const matrix3d_t inertia_matrix_inv = {
+    .array = {{1 / 0.17, 0, 0}, {0, 1 / 11.0, 0}, {0, 0, 1 / 11.0}}
+}; // inverse of inertia matrix of the rocket (J^-1 in matlab)
+// gravitational acceleration in the geographic inertial frame (g in matlab)
+static vector3d_t grav_acc = {.array = {-9.81, 0, 0}};
+
+// actuator and airfoil
+static const float tau_cl_alpha = 2.0f; // time constant to converge Cl back to 1.5 in filter
+static const float cl_alpha = 5.0f; // estimated coefficient of lift, const with Ma
+static const float tau = 1 / 20.0f; // time constant of first order actuator dynamics
+
+// initialize
+
+// timestamp
+#define MS_TO_SECONDS 1000.0
+static float prev_timestamp_ms = 0.0, current_timestamp_ms = 0.0; // timestamp of function call
+static double dt =
+    0.0; // in seconds, using double for precision (only in model_dynamic_update and jacobians)
+
+/*
+ * Dynamics update
+ * returns the new integrated state
+ */
+x_state_t model_dynamics_update(x_state_t *state, u_dynamics_t *input) {
```

Comment:
Couldn't hurt to pass as `const *`. Lets the compiler catch more potential mistakes